### DISCUSSION TITLE
Temperature Variable in Solidification

### DISCUSSION THREAD

**Discussion by rl3fz** [idaholab/moose#18159] [Q&A Modules: Phase field] [answered]
Hello everyone,

Is there a way to incorporate both undercooling and a thermal gradient in solidification modeling? I'm tinkering with GrandPotentialPlanarGrowth.i and GrandPotentialSolidification.i as a start, but I'm not sure which is more appropriate.

Any advice would be greatly appreciated!

(Edit: For reference https://github.com/idaholab/moose/tree/next/modules/phase_field/examples/anisotropic_interfaces)

**Comment by laagesen** [comment, marked as answer]
Hi there, it sounds like you are interested in alloy solidification rather than pure material solidification. In alloy solidification, typically solute diffuses much more slowly than heat, so temperature can be considered fixed by constraints external to the system and the latent heat of solidification can be neglected.

In that case, a fixed amount of undercooling (solidification at a constant temperature) can be accounted by setting the free energies for the solid and liquid phases to the appropriate values based on the system temperature. A temperature gradient can be included by making the free energies of the solid and liquid phases a function of position along the solidification direction- so the thermal gradient determines the temperature profile, which we do not solve for as a field variable. Instead we take that temperature as a function of position and modify the free energy of the solid and liquid phases as a function of temperature. To implement this, you could create an AuxVariable for temperature, and use a Function to set the temperature as a function of position. Then you could use a DerivativeParsedMaterial to make the solid and liquid phase free energies a function of the temperature (and if you are using one of the grand potential models, you'll need to convert from free energy density to grand potential density).

**Reply by rl3fz** [reply]
I'll try this, thank you!
